tests/kernel/install_kotd.pm

```diff
@@ -102,14 +103,25 @@ END
 
 sub run {
     my $self = shift;
+    my $default_version;
+    my $baseurl;
+
     $self->wait_boot;
 
     $self->select_serial_terminal;
 
-    my $url = get_required_var("INSTALL_KOTD");
+    if (is_sle) {
+        $default_version = "SLE" . get_required_var("VERSION");
+        $baseurl = "http://download.suse.de/ibs/Devel:";
+    } else {
+        $default_version = "HEAD";
+        $baseurl = "http://download.opensuse.org/repositories";
+    }
+
+    my $url = get_var("KOTD_REPO", $default_version);
     if ($url !~ /^https?:\/\//) {
```

Comment:
I would suggest removing this regex for $url and doing something like:

    if ! (my $url=get_var("KOTD_REPO")) {
        # generate the $url here
    } 

(not sure about syntax)